packages/components/src/dropdown-menu/index.js

```diff
@@ -64,26 +67,34 @@ function DropdownMenu( {
 						role="menu"
 						aria-label={ menuLabel }
 					>
-						{ controls.map( ( control, index ) => (
-							<IconButton
-								key={ index }
-								onClick={ ( event ) => {
-									if ( control.isDisabled ) {
-										return;
-									}
-									event.stopPropagation();
-									onClose();
-									if ( control.onClick ) {
-										control.onClick();
-									}
-								} }
-								className="components-dropdown-menu__menu-item"
-								icon={ control.icon }
-								role="menuitem"
-								disabled={ control.isDisabled }
-							>
-								{ control.title }
-							</IconButton>
+						{ flatMap( controlSets, ( controlSet, indexOfSet ) => (
+							controlSet.map( ( control, indexOfControl ) => (
+								<IconButton
+									key={ [ indexOfSet, indexOfControl ].join() }
+									onClick={ ( event ) => {
+										if ( control.isDisabled ) {
```

Comment:
Why should we expect `onClick` callback would ever be triggered when `disabled` prop is passed?

Speaking more from a high-level; this doesn't seem like it should happen.

**Edit:** Observing this is a hold-over from previous implementation. My concern still stands, but not necessarily within scope of the pull request.